aes-siv/src/siv.rs

```diff
@@ -227,15 +227,15 @@ where
     pub fn encrypt_inout_detached<I, T>(
         &mut self,
         headers: I,
-        plaintext: InOutBuf<'_, '_, u8>,
+        mut plaintext: InOutBuf<'_, '_, u8>,
     ) -> Result<Tag, Error>
     where
         I: IntoIterator<Item = T>,
         T: AsRef<[u8]>,
     {
         // Compute the synthetic IV for this plaintext
-        let siv_tag = s2v(&mut self.mac, headers, plaintext)?;
-        self.xor_with_keystream(siv_tag, plaintext);
+        let siv_tag = s2v(&mut self.mac, headers, plaintext.get_in())?;
+        self.xor_with_keystream(siv_tag, plaintext.get_out());
```

Comment:
This probably needs to pass `plaintext` and the receiver needs to load plaintext from in and write to out